components/tree/nz-tree-node.component.ts

```diff
@@ -275,6 +275,13 @@ export class NzTreeNodeComponent implements OnInit, OnChanges, OnDestroy {
         this.nzTreeNode.isLoading = true;
       }
       this.nzTreeNode.isExpanded = !this.nzTreeNode.isExpanded;
+      const nzTreeNodeChildren = this.nzTreeNode.getChildren();
+      for (const child of nzTreeNodeChildren) {
+        if (child.isMatched) {
+          this.nzTreeNode.isMatched = true;
```

Comment:
You should't set the parent node (`isMatched` equals false) property `isMatched ` to be true even it has matched child nodes, I think it would better to add an extra property like `private canHide` instead of setting `isMatched` to be true

Reply:
and you need to run `nz-tree.spec.ts` with coverage to check your tests.

Reply:
Thanks for review, I'll check later. For now I must focus for my wallet. :>